athena-dynamodb/athena-dynamodb.yaml

```diff
@@ -70,6 +78,8 @@ Resources:
                 - glue:GetDatabase
                 - athena:GetQueryExecution
                 - s3:ListAllMyBuckets
+                - kms:GenerateDataKey
+                - kms:GenerateRandom
```

Comment:
This is too broad. Can we add a conditional policy that grants this only on the specified key?

Reply:
Ok good point, addressed it. I had to explicitly define a role and policies instead of using the Policy property of connector function which does not support conditional definition (if KMSKeyId is not defined). It is a bit longer, mainly because I had to expand the SAM S3CrudPolicy but I believe it is the only clean way to do it.

Reply:
looks good now